src/CDN/CloudFront.php

```diff
@@ -265,21 +281,19 @@ private function getClient(): CloudFrontClient
                 $config['version'] = $this->version;
             }
 
-            $credentials = [
-                'key' => $this->region,
-                'secret' => $this->version,
-            ];
-
-            // @todo: Remove the following check and the `else` block when support for aws/aws-sdk-php < 3.0 is dropped.
-            if (class_exists(Sdk::class)) {
-                // AWS v3.x.
-                $this->client = new CloudFrontClient($config + ['credentials' => $credentials]);
-            } else {
-                // AWS v2.x.
-                $this->client = CloudFrontClient::factory($config + $credentials);
-            }
+            $this->client = CloudFrontClient::factory($config);
         }
+    }
 
-        return $this->client;
+    /**
+     * Generates a valid caller reference from given paths regardless its order.
+     *
+     * @return string a md5 representation
+     */
+    protected function getCallerReference(array $paths)
```

Comment:
```suggestion
     *
     * @phpstan-param list<string> $paths
     *
     * @return string a md5 representation
     */
    private function getCallerReference(array $paths): string
```

If it can't be private, let's make it `final`?

Reply:
The whole class is already marked as final.